application/helpers/export_helper.php

```diff
@@ -760,7 +760,15 @@ function buildXMLFromQuery($xmlwriter, $Query, $tagname = '', $excludes = array(
                 $result[] = $row->decrypt()->attributes;
             }
         } else {
-            $QueryResult = Yii::app()->db->createCommand($Query)->limit($iChunkSize, $iStart)->query();
+            /** @var CDbConnection $db */
+            $db = Yii::app()->db;
+            if(is_string($Query)) {
```

Comment:
Isn't $Query expected to be a string every time?

Reply:
no, mostly seems to be an array, but is sting in some cases. The ->limit() method only works while query is an array, see docs here:
https://www.yiiframework.com/doc/api/1.1/CDbCommand#__construct-detail
![image](https://github.com/LimeSurvey/LimeSurvey/assets/6357451/2c96012b-a34c-4c14-86b8-67823816b6f7)

if query is a string - calling ->limit() has no effect (hence the pr).